server/ServerFSM.cpp

```diff
@@ -473,6 +492,18 @@ bool ServerFSM::EstablishPlayer(const PlayerConnectionPtr& player_connection,
     return client_type != Networking::INVALID_CLIENT_TYPE;
 }
 
+std::string ServerFSM::GetAutoSaveFileName() {
+    std::string subdir = GetGalaxySetupData().GetGameUID();
+    boost::filesystem::path autosave_dir_path = GetServerSaveDir() / (subdir.empty() ? "auto" : subdir);
+    const auto& extension = MP_SAVE_FILE_EXTENSION;
+    // Add timestamp to autosave generated files
+    std::string datetime_str = FilenameTimestamp();
+
+    std::string save_filename = boost::io::str(boost::format("FreeOrion_%04d_%s%s") % m_server.CurrentTurn() % datetime_str % extension);
```

Comment:
is accessing `m_server.CurrentTurn()` the only reason this function is a member of `ServerFSM`?

Reply:
No, this code to generate save file name uses in two places now so it is also refactored to reduce code repetition. `ServerFSM` is a common space for both usages.

Reply:
my point was whether that was the only reason for it not to be a file-scope free function that doesn't appear in the header

Reply:
Fixed.